Library/Homebrew/exceptions.rb

```diff
@@ -70,16 +70,12 @@ class TapFormulaAmbiguityError < RuntimeError
 
   def initialize name, paths
     @name = name
-    @paths = paths
-    @formulae = paths.map do |path|
-      path.to_s =~ HOMEBREW_TAP_PATH_REGEX
-      "#{$1}/#{$2.sub("homebrew-", "")}/#{path.basename(".rb")}"
-    end
+    @paths = paths.sort! {|a, b| a.tap_priority <=> b.tap_priority}
 
     super <<-EOS.undent
-      Formulae found in multiple taps: #{formulae.map { |f| "\n       * #{f}" }.join}
+      Formulae found in multiple taps: #{@paths.map { |p| "\n       * [#{p.tap_priority.to_s.rjust(2, '0')}] #{p.fully_qualified_name}" }.join}
```

Comment:
This should probably instead be reworked so it's only ambiguous if you have two taps with the same priority.